inspirehep/modules/refextract/tasks.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Refextract tasks."""
+
+from __future__ import absolute_import, division, print_function
+
+import contextlib
+
+from celery import shared_task
+import codecs
+
+from invenio_db import db
+
+
+@contextlib.contextmanager
+def file_open_for_editing(fpath):
+    fh = codecs.open(fpath, encoding='utf-8', mode='w')
+
+    yield fh
+
+    fh.close()
+
+
+@shared_task()
+def create_journal_kb_file(path):
+    querystring = "SELECT short_titles, title_variants " \
```

Comment:
you need to sort the rows on the length of `title_variants` from longest to shortest as in https://github.com/inspirehep/refextract/blob/master/refextract/references/kbs/journal-titles.kb, in order to prevent wrong partial matches.